File: backend/Shared/EventPlanR.Domain/Entities/Order.cs
using EventPlanr.Domain.Common;
using EventPlanr.Domain.Enums;

namespace EventPlanr.Domain.Entities;

public class Order : BaseAuditableEntity
{
    public string CustomerUserId { get; set; } = null!;
    public string CustomerFirstName { get; set; } = null!;
    public string CustomerLastName { get; set; } = null!;
    public Address BillingAddress { get; set; } = null!;
    public Currency Currency { get; set; }
    public List<SoldTicket> SoldTickets { get; set; } = new List<SoldTicket>();
}

File: frontend/lib/domain/models/order/order.dart
import 'package:event_planr_app/domain/models/common/address.dart';
import 'package:event_planr_app/domain/models/event/currency_enum.dart';
import 'package:event_planr_app/domain/models/ticket/sold_ticket.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'order.freezed.dart';

@freezed
class Order with _$Order {
  const factory Order({
    required String customerFirstName,
    required String customerLastName,
    required Address billingAddress,
    required double total,
    required CurrencyEnum currency,
    required List<SoldTicket> soldTickets,
  }) = _Order;
}
